pkg/scheduler/api/v1/types.go

```diff
@@ -158,7 +158,7 @@ type ExtenderArgs struct {
 	Pod apiv1.Pod `json:"pod"`
 	// List of candidate nodes where the pod can be scheduled; to be populated
 	// only if ExtenderConfig.NodeCacheCapable == false
-	Nodes *apiv1.NodeList `json:"nodes,omitempty"`
+	Nodes []*apiv1.Node `json:"nodes,omitempty"`
```

Comment:
doesn't this change the format sent to the extender? is that a breaking change?

Reply:
AFAIK, only @ravigadde is using the feature, and we did not commit backward compatibility before.

Reply:
it is documented in https://kubernetes.io/docs/concepts/overview/extending/#scheduler-extensions and https://github.com/kubernetes/community/blob/master/contributors/design-proposals/scheduling/scheduler_extender.md without any statements that it is unsupported. The API version of the data sent is v1, which also doesn't indicate it is alpha level or unsupported.

Reply:
This is a breaking change and the benefits are unclear to me.

Reply:
@liggitt I'm afraid extender is neither versioned nor feature gated. So this is expect to be a break change. 

The intention is to eliminate all `nodeList.Items` mess.

But I'm totally fine to leave this as is if there's already user of it.  As this part will be refactored eventually when scheduler is reworked to framework. cc @k82cn

Reply:
the data sent to the extender is versioned, right?

Reply:
Right, but I think it cannot help solve compatibility problem. The existing extenders will be break after upgrade. Unless we send announcement and think it is tolerable,

Reply:
if you wanted to change the structure, you'd add a version of the extender API and let extenders specify in the config what version they wanted sent to them (defaulting to v1)

Reply:
Bumping up the version for a format change which doesn't really compress the data or add any additional features is not worth it imo. If we need to bump the version because of a new feature, we could probably take this up at that time?

Reply:
Ya, while I'm afraid the effort of maintaining version would be much more than benefits it brings.  It also confused me a little bit that we do have apiVersion in the config: https://github.com/kubernetes/kubernetes/blob/master/examples/scheduler-policy-config-with-extender.json#L20 while looks like not supported.

Reply:
I've explained the same concen :) @ravigadde While can you confirm what happened to the config example? It's out-of-date or mistaken?

Reply:
Thanks for keeping me honest @resouer Originally we had API version here, but before the release (1.2), it was removed as south bound versioning was not clear. The example is out of date, I can submit a PR to fix it.

Reply:
Yes, please go ahead. I think remove the `apiVersion` would be the fix.

Reply:
Thanks @resouer Submitted #58480

Reply:
> the effort of maintaining version would be much more than benefits it brings

When we make "scheduler as framework/platform" as Bobby suggested, we need version.

Reply:
@k82cn Honestly speaking, still no clear time frame. We may try to help push it somehow.

Reply:
yes; it's also a point of that feature. We need to put more effort in extender version and put backward compatibility as high priority.

Reply:
I guess we don't even need extender if scheduler can be re-worked to a framework.

Reply:
maybe, let discuss this in this design doc (gRPC part).